### PR TITLE
add basic test

### PR DESCRIPTION
**Pull request by mcalus3**
Added tests but there is still an error. The data is fetched correctly, so I predict that it's related to the ability of displaying backstage components in jest environment.

Edit: it turned out rendered component had to be wrapped in the ThemeProvider and Router

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,9 +29,12 @@
     "@material-ui/core": "^4.9.1",
     "@material-ui/icons": "^4.9.1",
     "@material-ui/lab": "4.0.0-alpha.45",
+    "@types/node-fetch": "^2.5.7",
+    "cross-fetch": "^3.0.6",
     "date-fns": "^2.15.0",
     "history": "^5.0.0",
     "moment": "^2.27.0",
+    "node-fetch": "^2.6.1",
```
Comment (iain-b) [inline package.json]: This and the @types aren't used, I think.

File: src/components/LastBuildCard.test.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+describe('LastBuildCard', () => {
+  const worker = setupServer();
+  msw.setupDefaultHandlers(worker);
+
+  beforeEach(() => {
+    // jest.resetAllMocks();
+    worker.use(
+      rest.get(
+        'http://exampleapi.com/travisci/api/repo/RoadieHQ%2Fsample-service/builds?offset=0&limit=1',
+        (_, res, ctx) => res(ctx.json(buildsResponseMock))
```
Comment (iain-b) [inline src/components/LastBuildCard.test.tsx:51]: This response has limit 5 no? I guess this is here as the user can decide to fetch only the most recent N builds and this would have a corresponding response?

File: src/components/LastBuildCard.test.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      <ThemeProvider theme={lightTheme}>
+        <MemoryRouter>
+          <ApiProvider apis={apis}>
+            <RecentTravisCIBuildsWidget entity={entityMock} />
+          </ApiProvider>
+        </MemoryRouter>
+      </ThemeProvider>
+    );
+    expect(
+      await rendered.findByText('Create catalog-info-5.yaml')
```
Comment (iain-b) [inline src/components/LastBuildCard.test.tsx]: Could we reference this value from the responseMock object? e.g. buildsResponseMock.builds[N].commit.message (or whatever the path is). It just makes it easier to understand what the code is doing as well as the test.

File: src/components/LastBuildCard.test.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        (_, res, ctx) => res(ctx.json(buildsResponseMock))
+      ),
+      rest.get(
+        'http://exampleapi.com/travisci/api/repo/RoadieHQ%2Fsample-service/builds?offset=0&limit=5',
+        (_, res, ctx) => res(ctx.json(buildsResponseMock))
+      )
+    );
+  });
+
+  it('should display widget with data', async () => {
```
Comment (iain-b) [inline src/components/LastBuildCard.test.tsx:60]: If possible could we add tests for: empty (0 builds), loading and error states?

Again if there's anything overly time consuming in that it can wait.

File: src/components/LastBuildCard.test.tsx (hunk 2 above)
Comment (mcalus3) [reply src/components/LastBuildCard.test.tsx:51]: For some reason when page loads, there are two requests going - one with limit 1and second with the limit set by the user. Could track it down and fix, but wanted only to add test in this pull request.

File: src/components/LastBuildCard.test.tsx (hunk 4 above)
Comment (mcalus3) [reply src/components/LastBuildCard.test.tsx:60]: Answered in:
https://github.com/RoadieHQ/backstage-plugin-security-insights/pull/8#discussion_r555761306

File: src/components/LastBuildCard.test.tsx (hunk 3 above)
Comment (mcalus3) [reply src/components/LastBuildCard.test.tsx]: Done

File: package.json (hunk 1 above)
Comment (mcalus3) [reply package.json]: Done

File: src/components/LastBuildCard.test.tsx (hunk 2 above)
Comment (iain-b) [reply src/components/LastBuildCard.test.tsx:51]: Interesting. Yep we can look into it separately.

**Review by iain-b — APPROVED** [review]
Thanks, Marek!
